### PR TITLE
chore: Fixes breaking change log with goreleaser

### PR DESCRIPTION
**Pull request by maastha**
## Description

Fixes breaking change log with goreleaser.

Generated test changlog: https://github.com/mongodb/atlas-sdk-go/blob/89fdc186ed4b2c1c1b3c95559bfab03d3f67a291/tools/releaser/breaking_changes/v20241113004.md

Link to any related issue(s): [CLOUDP-281143](https://jira.mongodb.org/browse/CLOUDP-281143)

## Type of change:

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added any necessary documentation (if appropriate)
- [ ] I have run `make fmt` and formatted my code

### CODE REVIEW

**Review by lantoli — DISMISSED** [review]

File: tools/releaser/scripts/breaking-changes.sh [hunk 1]
```diff
@@ -1,37 +1,45 @@
...
 
-echo "Installing go-apidiff"
-go install github.com/joelanford/go-apidiff@latest > /dev/null
+# shellcheck source=/dev/null
+source "$script_path/extract-version.sh"
+BASE_VERSION="github.com/mongodb/atlas-sdk-go/$SDK_MAJOR_VERSION@$SDK_VERSION"
 
-echo "Running breaking changes check comparing commits ${API_DIFF_OLD_COMMIT} and ${API_DIFF_NEW_COMMIT}"
+echo "Installing gorelease"
+go install golang.org/x/exp/cmd/gorelease@latest >/dev/null
```
Comment (wtrocki) [inline tools/releaser/scripts/breaking-changes.sh:13]: :FYI: Latest golang does have tools directive that allows us to define tools direclty in the go.mod.
See: https://github.com/golang/go/issues/48429#issuecomment-2547760589

File: tools/releaser/scripts/breaking-changes.sh [hunk 2]
```diff
@@ -1,37 +1,45 @@
 #!/bin/bash
 set -eu
-GOPATH=$(go env GOPATH)
 
 # Inputs:
-# API_DIFF_OLD_COMMIT: commit before the API changes to compare with. If not provided, script will fail with "unbound variable" error
-# API_DIFF_NEW_COMMIT: commit with the new API changes. If not provided, script will fail with "unbound variable" error
-# TARGET_BREAKING_CHANGES_FILE - file to save breaking changes
```
Comment (wtrocki) [inline tools/releaser/scripts/breaking-changes.sh:8]: Does script do not have any inputs right now?

**Comment by wtrocki** [comment]
Script apart from printing breaking changes to file is also responsible in detection of major breaking change. 
I asume that we did local tests covering this use case?

**Review by wtrocki — DISMISSED** [review]
LGTM 

Consider updating documentation to mention what tools we use etc.

File: tools/releaser/scripts/breaking-changes.sh (hunk 2 above)
Comment (maastha) [reply tools/releaser/scripts/breaking-changes.sh:8]: no.  The script now essentially uses the last release version (from extract-version.sh)

**Comment by maastha** [comment]
>Script apart from printing breaking changes to file is also responsible in detection of major breaking change.
I asume that we did local tests covering this use case?

right. Detection of breaking changes is evaluated by whether `BREAKING_CHANGES` is empty or not, that logic has not been updated and tested for no breaking changes case as well.

**Comment by wtrocki** [comment]
Super nice work @maastha Good to merge as we going to see potential new SDK releases today!

**Review by wtrocki — APPROVED** [review]

**Pull request merged**